src/state/assets/userAssets.ts

```diff
@@ -156,7 +157,8 @@ export const createUserAssetsStore = (address: Address | string) =>
         const queryKey = getSearchQueryKey({ filter, searchQuery: inputSearchQuery });
 
         // Use an external function to get the cache to prevent updates in response to changes in the cache
-        const cachedData = getCurrentSearchCache()?.get(queryKey);
+        const currentSearchCache = getCurrentSearchCache();
+        const cachedData = !isEmpty(currentSearchCache) ? getCurrentSearchCache()?.get(queryKey) : [];
```

Comment:
need to look into this further, but this was returning a `{}` instead of a `Map` sometimes, which was causing a crash when it couldn't find `?.get` as a function

Reply:
i think this has to do with userAssetsStore persistence. it looks like `searchCache` serialization/deserialization isn't handled explicitly, which would cause store hydration to be messed up since `Map`s aren't JSON-compatible.

https://github.com/rainbow-me/rainbow/blob/develop/src/state/assets/userAssets.ts#L61-L66

Reply:
also not sure if we intend for the entire store to be persisted

Reply:
Yeah we have no persistence right now because we're failing to serialize. This needs to be fixed before anything else. I don't think we need to persist the search cache fwiw.

Reply:
i think (?) it does actually persist, but it gets deserialized in an unexpected format (not converted back to `Map`)

Reply:
Sorry I meant we're never actually persisting any data. The only thing that gets persisted is `{}`